logbook-core/src/main/java/org/zalando/logbook/ChunkingSink.java

```diff
@@ -0,0 +1,54 @@
+package org.zalando.logbook;
+
+import org.apiguardian.api.API;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+
+import static java.util.stream.StreamSupport.stream;
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+import static org.zalando.fauxpas.FauxPas.throwingConsumer;
+
+@API(status = EXPERIMENTAL)
+public final class ChunkingSink implements Sink {
+
+    private static final int MIN_MAX_DELTA = 16;
+
+    private final Sink delegate;
+    private final int minChunkSize;
+    private final int maxChunkSize;
+
+    public ChunkingSink(final Sink delegate, final int size) {
+        this.delegate = delegate;
+
+        if (size <= 0) {
+            throw new IllegalArgumentException("size is expected to be greater than zero");
+        }
+        this.minChunkSize = size > MIN_MAX_DELTA ? size - MIN_MAX_DELTA : size;
+        this.maxChunkSize = size;
+    }
+
+    @Override
+    public boolean isActive() {
+        return delegate.isActive();
+    }
+
+    @Override
+    public void write(final Precorrelation precorrelation, final HttpRequest original) throws IOException {
+        chunk(original)
+                .map(chunk -> new BodyReplacementHttpRequest(original, chunk))
+                .forEach(throwingConsumer(replaced -> delegate.write(precorrelation, replaced)));
+    }
+
+    @Override
+    public void write(final Correlation correlation, final HttpRequest request, final HttpResponse original) throws IOException {
+        chunk(original)
+                .map(chunk -> new BodyReplacementHttpResponse(original, chunk))
+                .forEach(throwingConsumer(replaced -> delegate.write(correlation, request, replaced)));
```

Comment:
What about going on step further and hide the headers, like

```
final Iterator<String> chunks = chunk(original).iterator();
final String firstChunk = chunks.next();
delegate.write(correlation, request, new BodyReplacementHttpResponse(original, firstChunk));
chunks.forEachRemaining(chunk -> delegate.write(correlation, request, new BodyReplacementHttpResponse(new HeaderHidingHttpResponse(original), chunk)));
```

Reply:
That would move it a bit closer to be a properly usable feature. Until now it was basically breaking our format, but even with my change it's still a gimmick and not 100% *"production-ready"*.

As an example a parser wouldn't know that this is a chunk. I believe it would be nice to get attribute-support in (see #381) and use that to *tag* individual chunks with their number. Parsers like Scalyr could then exclude anything with `$chunk > 1` when counting requests/responses (as an example).